### ISSUE TITLE
[mihome] Single press on wireless switch is reported as dobule press

### ISSUE THREAD

**Issue by Masssssy** [openhab/openhab2-addons#4136]
## Expected Behavior

One press on both the buttons should register as one dobule press not two. One button press on one of the buttons should register as one press, not one press and one unknown event.

If this button (WXKG02LM / 86sw2) supports other types of presses (such as long press) these do not work.

It seems that the binding recognizes a long click no matter how quick the button is pressed but it's reported as something else as seen in the events.log. I'm 50% that this button does not even support long press.

## Current Behavior

* Single press reports an extra event as "UNKNOWN_EVENT" after "SHORT_PRESSED"
* Double press reports "SHORT_PRESSED" twice in a row. 

Double press is hence completely unusable. I'm also not sure if the buttons support LONG_PRESS etc, if that's the case these do not work either. 

**Pressing a single button**

**events.log**
```
2018-10-14 21:03:33.718 [vent.ChannelTriggeredEvent] - mihome:86sw2:158d00026b9653:ch1 triggered SHORT_PRESSED
2018-10-14 21:03:33.722 [vent.ChannelTriggeredEvent] - mihome:86sw2:158d00026b9653:ch1 triggered UNKNOWN_EVENT

```

**openhab.log**
```
2018-10-14 20:49:42.214 [DEBUG] [.mihome.internal.socket.XiaomiSocket] - Received Datagram from 192.168.1.218:4321 on Port 9898
2018-10-14 20:49:42.214 [TRACE] [g.mihome.handler.XiaomiBridgeHandler] - Received message {"cmd":"report","model":"86sw2","sid":"158d00026b9653","short_id":65473,"data":"{\"channel_0\":\"click\"}"}
2018-10-14 20:49:42.214 [TRACE] [g.mihome.handler.XiaomiBridgeHandler] - Updated "last time seen" for device 158d00026b9653
2018-10-14 20:49:42.214 [DEBUG] [home.handler.XiaomiDeviceBaseHandler] - Item got update: {"cmd":"report","model":"86sw2","sid":"158d00026b9653","short_id":65473,"data":"{\"channel_0\":\"click\"}"}
2018-10-14 20:49:42.215 [TRACE] [.mihome.internal.socket.XiaomiSocket] - Data received and notified 1 listeners
2018-10-14 20:49:42.215 [TRACE] [.mihome.internal.socket.XiaomiSocket] - Thread Thread[Thread-92,5,main] waiting for data on port 9898
2018-10-14 20:49:42.218 [DEBUG] [.mihome.internal.socket.XiaomiSocket] - Received Datagram from 192.168.1.218:4321 on Port 9898
2018-10-14 20:49:42.221 [TRACE] [g.mihome.handler.XiaomiBridgeHandler] - Received message {"cmd":"report","model":"86sw2","sid":"158d00026b9653","short_id":65473,"data":"{\"channel_0\":\"long_click\"}"}
2018-10-14 20:49:42.221 [TRACE] [g.mihome.handler.XiaomiBridgeHandler] - Updated "last time seen" for device 158d00026b9653
2018-10-14 20:49:42.221 [DEBUG] [home.handler.XiaomiDeviceBaseHandler] - Item got update: {"cmd":"report","model":"86sw2","sid":"158d00026b9653","short_id":65473,"data":"{\"channel_0\":\"long_click\"}"}
2018-10-14 20:49:42.221 [TRACE] [.mihome.internal.socket.XiaomiSocket] - Data received and notified 1 listeners
2018-10-14 20:49:42.222 [TRACE] [.mihome.internal.socket.XiaomiSocket] - Thread Thread[Thread-92,5,main] waiting for data on port 9898
```

**Double Press**

events.log
```
2018-10-14 21:04:04.621 [vent.ChannelTriggeredEvent] - mihome:86sw2:158d00026b9653:dual_ch triggered SHORT_PRESSED
2018-10-14 21:04:04.630 [vent.ChannelTriggeredEvent] - mihome:86sw2:158d00026b9653:dual_ch triggered SHORT_PRESSED
```

openhab.log
```
2018-10-14 20:51:11.743 [DEBUG] [.mihome.internal.socket.XiaomiSocket] - Received Datagram from 192.168.1.218:4321 on Port 9898
2018-10-14 20:51:11.743 [TRACE] [g.mihome.handler.XiaomiBridgeHandler] - Received message {"cmd":"report","model":"86sw2","sid":"158d00026b9653","short_id":65473,"data":"{\"dual_channel\":\"both_click\"}"}
2018-10-14 20:51:11.743 [TRACE] [g.mihome.handler.XiaomiBridgeHandler] - Updated "last time seen" for device 158d00026b9653
2018-10-14 20:51:11.743 [DEBUG] [home.handler.XiaomiDeviceBaseHandler] - Item got update: {"cmd":"report","model":"86sw2","sid":"158d00026b9653","short_id":65473,"data":"{\"dual_channel\":\"both_click\"}"}
2018-10-14 20:51:11.743 [TRACE] [.mihome.internal.socket.XiaomiSocket] - Data received and notified 1 listeners
2018-10-14 20:51:11.743 [TRACE] [.mihome.internal.socket.XiaomiSocket] - Thread Thread[Thread-92,5,main] waiting for data on port 9898
2018-10-14 20:51:11.750 [DEBUG] [.mihome.internal.socket.XiaomiSocket] - Received Datagram from 192.168.1.218:4321 on Port 9898
2018-10-14 20:51:11.750 [TRACE] [g.mihome.handler.XiaomiBridgeHandler] - Received message {"cmd":"report","model":"86sw2","sid":"158d00026b9653","short_id":65473,"data":"{\"dual_channel\":\"long_both_click\"}"}
2018-10-14 20:51:11.750 [TRACE] [g.mihome.handler.XiaomiBridgeHandler] - Updated "last time seen" for device 158d00026b9653
2018-10-14 20:51:11.751 [DEBUG] [home.handler.XiaomiDeviceBaseHandler] - Item got update: {"cmd":"report","model":"86sw2","sid":"158d00026b9653","short_id":65473,"data":"{\"dual_channel\":\"long_both_click\"}"}
2018-10-14 20:51:11.751 [TRACE] [.mihome.internal.socket.XiaomiSocket] - Data received and notified 1 listeners
2018-10-14 20:51:11.751 [TRACE] [.mihome.internal.socket.XiaomiSocket] - Thread Thread[Thread-92,5,main] waiting for data on port 9898
```

## Steps to Reproduce
To reproduce connect the button (WXKG02LM) to the Xiaomi mi home gateway and then add it as a thing and respective items in Openhab2. 

## Your Environment
* Version used: (e.g., openhab and addon versions)
Openhab 2.3 Stable - binding-mihome - 2.3.0
* Environment name and version (e.g. Chrome 39, node.js 5.4, Java 8, ...):
java version "1.8.0_181"
* Operating System and version (desktop or mobile, Windows 10, Raspbian Jessie, ...):
Ubuntu 16.04

**Comment by Masssssy** [comment]
Great!

I have just tested it. It now reports SHORT_PRESSED followed by LONG_PRESSED which is better functionality wise but still seems incorrect. I guess it is OK if it was an intentional choice but I feel it would be better to just get one event if possible.

**Comment by dimalo** [comment]
Yeah that is by intent. I am not sure if it's a bug and will be fixed in later gateway firmware versions. So let's have it in and let the user choose which event he wants to trigger on.
I understand it's a bit strange to receive two events for one action, but that is what the gateway actually sends.
I would like to introduce workarounds only where otherwise the binding does not work correctly. This behaviour is ignoreable in my opinion and maybe in the future the binding will trigger on real long presses correctly instead of needing a fix again because it's ignoring them.

**Comment by cweitkamp** [comment]
Closed by #4137
